components/txn_types/src/lock.rs

```diff
@@ -128,6 +194,8 @@ impl Lock {
         let mut for_update_ts = TimeStamp::zero();
         let mut txn_size: u64 = 0;
         let mut min_commit_ts = TimeStamp::zero();
+        let mut use_async_commit = false;
+        let mut secondaries = Vec::new();
```

Comment:
Very small suggestion: use `secondaries: Option<Vec<_>>`, since it is an invariant of the code that `use_async_commit == true` if and only if secondaries has been set.